frontend/src/lib/OutputArtifactLoader.ts

```diff
@@ -198,6 +217,275 @@ export class OutputArtifactLoader {
     };
   }
 
+  public static async getMlmdContext(argoPodName: string): Promise<Context> {
+    if (argoPodName.split('-').length < 3) {
+      throw new Error('kfpPodName has fewer than 3 parts');
+    }
+
+    // argoPodName has the general form "pipelineName-workflowId-executionId".
+    // All components of a pipeline within a single run will have the same
+    // "pipelineName-workflowId" prefix.
+    const pipelineName = argoPodName
+      .split('-')
+      .slice(0, -2)
+      .join('_');
+    const runID = argoPodName
+      .split('-')
+      .slice(0, -1)
+      .join('-');
+    const contextName = pipelineName + '.' + runID;
+
+    const request = new GetContextByTypeAndNameRequest();
+    request.setTypeName('run');
+    request.setContextName(contextName);
+    let res: GetContextByTypeAndNameResponse;
+    try {
+      res = await Api.getInstance().metadataStoreService.getContextByTypeAndName(request);
+    } catch (err) {
+      err.message = 'Failed to getContextsByTypeAndName: ' + err.message;
+      throw err;
+    }
+
+    const context = res && res.getContext();
+    if (!context) {
+      throw new Error("getContextByTypeAndName didn't have a context");
+    }
+    return context;
+  }
+
+  public static async getExecutionInContextWithPodName(
+    argoPodName: string,
+    context: Context,
+  ): Promise<Execution> {
+    const contextId = context.getId();
+    if (!contextId) {
```

Comment:
I'm not sure, double checking if this is what you meant.

context.getId()'s type is already number, so it couldn't be null.
if contextId === 0, it's an invalid value. Is that what you mean? Probably directly checking whether contextId is positive would be more readable if that's what you meant.